### ISSUE TITLE
Make jekyll/jekyll code compliant with rubocop rules

### ISSUE THREAD

**Issue by parkr** [jekyll/jekyll#4885]
Rubocop can help us keep our code consistent in style. Consistency makes it easier to read the code and having Rubocop handle reporting offenses makes feedback to pull requests quick & less controversial.

Instructions for fixing a file:
1. Ensure there isn't an existing pull request fixing the files you're looking at updating
2. Remove the file from `.rubocop.yml` under `AllCops.Exclude`.
3. Run `bundle install` to ensure you have all the gems.
4. Run `script/fmt`.
5. Fix errors until `script/fmt` reports no errors.
6. Commit to a feature branch on your fork and create a pull request mentioning this issue.
7. Rejoice!

Here's a checklist of files **in-progress**:
- [x] Fix offenses in `lib/jekyll/convertible.rb` and enable in `.rubocop.yml` (#5031, #5308)
- [x] Fix offenses in `lib/jekyll/document.rb` and enable in `.rubocop.yml` (#5030, #5045)
- [x] Fix offenses in `lib/jekyll/collection.rb` and enable in `.rubocop.yml` (#5022)
- [x] Fix offenses in `lib/jekyll/configuration.rb` and enable in `.rubocop.yml` (#5024)
- [x] Fix offenses in `lib/jekyll/regenerator.rb` and enable in `.rubocop.yml` (#5025)
- [ ] Fix offenses in `lib/jekyll/renderer.rb` and enable in `.rubocop.yml` (#5052)
- [x] Fix offenses in `lib/jekyll/utils.rb` and enable in `.rubocop.yml` (#5026)

Here's a checklist of completed tasks:
- [x] Finalize Rubocop rules (#4886)
- [x] Enable Rubocop check for pull requests (#4886)
- [x] Fix offenses in `exe/jekyll` and enable in `.rubocop.yml` (#5017)
- [x] Fix offenses in `lib/jekyll/cleaner.rb` and enable in `.rubocop.yml` (#4892)
- [x] Fix offenses in `lib/jekyll/commands/build.rb` and enable in `.rubocop.yml` (#4888)
- [x] Fix offenses in `lib/jekyll/commands/clean.rb` and enable in `.rubocop.yml` (#4888)
- [x] Fix offenses in `lib/jekyll/commands/doctor.rb` and enable in `.rubocop.yml` (#4888)
- [x] Fix offenses in `lib/jekyll/commands/help.rb` and enable in `.rubocop.yml` (#4888)
- [x] Fix offenses in `lib/jekyll/commands/new.rb` and enable in `.rubocop.yml` (#4888)
- [x] Fix offenses in `lib/jekyll/commands/serve/servlet.rb` and enable in `.rubocop.yml` (#4888)
- [x] Fix offenses in `lib/jekyll/commands/serve.rb` and enable in `.rubocop.yml` (#4888)
- [x] Fix offenses in `lib/jekyll/command.rb` and enable in `.rubocop.yml` (#5018)
- [x] Fix offenses in `lib/jekyll/converter.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/converters/identity.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/converters/markdown` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/converters/markdown/kramdown_parser.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/converters/markdown/rdiscount_parser.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/converters/markdown/redcarpet_parser.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/converters/markdown.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/converters/smartypants.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/deprecator.rb` and enable in `.rubocop.yml` (#4940)
- [x] Fix offenses in `lib/jekyll/drops/collection_drop.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/drops/document_drop.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/drops/drop.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/drops/jekyll_drop.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/drops/site_drop.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/drops/unified_payload_drop.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/drops/url_drop.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/entry_filter.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/errors.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/excerpt.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/external.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/filters.rb` and enable in `.rubocop.yml` (#4993)
- [x] Fix offenses in `lib/jekyll/frontmatter_defaults.rb` and enable in `.rubocop.yml` (#4974)
- [x] Fix offenses in `lib/jekyll/generator.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/hooks.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/layout.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/liquid_extensions.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/liquid_renderer/file.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/liquid_renderer/table.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/liquid_renderer.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/log_adapter.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/page.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/plugin.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/plugin_manager.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/publisher.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/reader.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/readers/collection_reader.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/readers/data_reader.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/readers/layout_reader.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/readers/page_reader.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/readers/post_reader.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/readers/static_file_reader.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/related_posts.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/renderer.rb` and enable in `.rubocop.yml` (#4301)
- [x] Fix offenses in `lib/jekyll/site.rb` and enable in `.rubocop.yml` (#4973)
- [x] Fix offenses in `lib/jekyll/static_file.rb` and enable in `.rubocop.yml` (#4301)
- [x] Fix offenses in `lib/jekyll/stevenson.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/tags/highlight.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/tags/include.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/tags/link.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/tags/post_url.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/theme.rb` and enable in `.rubocop.yml` (#4959)
- [x] Fix offenses in `lib/jekyll/url.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `lib/jekyll/utils/ansi.rb` and enable in `.rubocop.yml` (#4916)
- [x] Fix offenses in `lib/jekyll/utils/platforms.rb` and enable in `.rubocop.yml` (#4916)
- [x] Fix offenses in `lib/jekyll/version.rb` and enable in `.rubocop.yml` (#4916)
- [x] Fix offenses in `lib/jekyll.rb` and enable in `.rubocop.yml` (#4966)
- [x] Fix offenses in `features/step_definitions.rb` and enable in `.rubocop.yml` (#4956)
- [x] Fix offenses in `features/support/formatter.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `features/support/helpers.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `test/helper.rb` and enable in `.rubocop.yml` (#4902)
- [x] Fix offenses in `test/simplecov_custom_profile.rb` and enable in `.rubocop.yml` (#4902)
- [x] Fix offenses in `test/test_ansi.rb` and enable in `.rubocop.yml` (#4902)
- [x] Fix offenses in `test/test_cleaner.rb` and enable in `.rubocop.yml` (#4902)
- [x] Fix offenses in `test/test_coffeescript.rb` and enable in `.rubocop.yml` (#4902)
- [x] Fix offenses in `test/test_collections.rb` and enable in `.rubocop.yml` (#4902)
- [x] Fix offenses in `test/test_command.rb` and enable in `.rubocop.yml` (#4902)
- [x] Fix offenses in `test/test_commands_serve.rb` and enable in `.rubocop.yml` (#4902)
- [x] Fix offenses in `test/test_configuration.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `test/test_convertible.rb` and enable in `.rubocop.yml` (#4902)
- [x] Fix offenses in `test/test_doctor_command.rb` and enable in `.rubocop.yml` (#4902)
- [x] Fix offenses in `test/test_document.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `test/test_entry_filter.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `test/test_excerpt.rb` and enable in `.rubocop.yml` (#4902)
- [x] Fix offenses in `test/test_filters.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `test/test_front_matter_defaults.rb` and enable in `.rubocop.yml` (#4902)
- [x] Fix offenses in `test/test_generated_site.rb` and enable in `.rubocop.yml` (#4902)
- [x] Fix offenses in `test/test_kramdown.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `test/test_layout_reader.rb` and enable in `.rubocop.yml` (#4902)
- [x] Fix offenses in `test/test_liquid_extensions.rb` and enable in `.rubocop.yml` (#4902)
- [x] Fix offenses in `test/test_liquid_renderer.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `test/test_log_adapter.rb` and enable in `.rubocop.yml` (#4902)
- [x] Fix offenses in `test/test_new_command.rb` and enable in `.rubocop.yml` (#4902)
- [x] Fix offenses in `test/test_page.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `test/test_path_sanitization.rb` and enable in `.rubocop.yml` (#4902)
- [x] Fix offenses in `test/test_plugin_manager.rb` and enable in `.rubocop.yml` (#4902)
- [x] Fix offenses in `test/test_rdiscount.rb` and enable in `.rubocop.yml` (#4902)
- [x] Fix offenses in `test/test_redcarpet.rb` and enable in `.rubocop.yml` (#4902)
- [x] Fix offenses in `test/test_regenerator.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `test/test_related_posts.rb` and enable in `.rubocop.yml` (#4916)
- [x] Fix offenses in `test/test_sass.rb` and enable in `.rubocop.yml` (#4916)
- [x] Fix offenses in `test/test_site.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `test/test_static_file.rb` and enable in `.rubocop.yml` (#4916)
- [x] Fix offenses in `test/test_tags.rb` and enable in `.rubocop.yml`
- [x] Fix offenses in `test/test_theme.rb` and enable in `.rubocop.yml` (#4916)
- [x] Fix offenses in `test/test_url.rb` and enable in `.rubocop.yml` (#4916)
- [x] Fix offenses in `test/test_utils.rb` and enable in `.rubocop.yml`

**Comment by brint** [comment]
#4902 has been merged, so those can be checked off.

**Comment by pathawks** [comment]
Fixing converters in #4931

**Comment by pathawks** [comment]
For those playing along at home, it _looks_ like the following are still up for grabs:

- `lib/jekyll/convertible.rb`
- `lib/jekyll/document.rb`
- `lib/jekyll/drops/site_drop.rb`
- `lib/jekyll/entry_filter.rb`
- `lib/jekyll/filters.rb`
- `lib/jekyll/frontmatter_defaults.rb`
- `lib/jekyll/layout.rb`
- `lib/jekyll/plugin_manager.rb`
- `lib/jekyll/regenerator.rb`
- `lib/jekyll/renderer.rb`
- `lib/jekyll/site.rb`
- `lib/jekyll/static_file.rb`
- `lib/jekyll/stevenson.rb`
- `lib/jekyll/theme.rb`
- `lib/jekyll/url.rb`
- `lib/jekyll/utils.rb`
- `lib/jekyll.rb`
- `features/step_definitions.rb`

Perhaps we can coordinate our efforts by posting here when we intend to work on a couple files? That way, we can avoid duplicating effort and (hopefully) minimize merge conflicts :+1:

**Comment by parkr** [comment]
W O W!!! Well done everyone, only a few more files to go.

**Comment by ayastreb** [comment]
I could take `lib/jekyll/convertible.rb`, `lib/jekyll/document.rb` and perhaps `features/step_definitions.rb` to fix today or tomorrow.

**Comment by ayastreb** [comment]
I've started working on [convertible.rb](https://github.com/jekyll/jekyll/blob/master/lib/jekyll/convertible.rb) and it looks like there is some code similar to [renderer.rb](https://github.com/jekyll/jekyll/blob/master/lib/jekyll/renderer.rb)
Some methods are exactly the same, others are a bit different.
What would be the better approach - trying to refactor out the duplication or just fix the code style in both files?

**Comment by derekgottlieb** [comment]
Opened #4959 to try to tackle the two remaining easy ones (theme and url libs).

**Comment by pathawks** [comment]
@ayastreb If you are able to DRY up the code, I vote for that :+1:

**Comment by ayastreb** [comment]
@pathawks ok, I will reserve `lib/jekyll/renderer.rb` for me then and try to refactor it together with `lib/jekyll/convertible.rb`

**Comment by parkr** [comment]
Only 8 more files to go! 🎉 I split the lists up so they would be easier to scan – top list is "not done," second list is "has a PR," and last is "complete."

**Comment by pathawks** [comment]
It looks like _everything_ is either finished, or in the process of being taken care of! :tada:

**Comment by jekyllbot** [comment]
This issue has been automatically marked as stale because it has not been commented on for at least two months.

The resources of the Jekyll team are limited, and so we are asking for your help.

If this is a **bug** and you can still reproduce this error on the <code>3.1-stable</code> or <code>master</code> branch, please reply with all of the information you have about it in order to keep the issue open.

If this is a **feature request**, please consider building it first as a plugin. Jekyll 3 introduced [hooks](http://jekyllrb.com/docs/plugins/#hooks) which provide convenient access points throughout the Jekyll build pipeline whereby most needs can be fulfilled. If this is something that cannot be built as a plugin, then please provide more information about why in order to keep this issue open.

This issue will automatically be closed in two months if no further activity occurs. Thank you for all your contributions.

**Comment by jekyllbot** [comment]
Hello! We are utilizing a new workflow in our issues and pull requests. Affinity teams have been setup to allow community members to hear about pull requests that may be interesting to them. When a new issue or pull request comes in, we are asking that the author mention the appropriate affinity team. I then assign a random "team captain" or two to the issue who is in charge of triaging it until it is closed or passing it off to another captain. In order to move forward with this new workflow, we need to know: which of the following teams best fits your issue or contribution?

- `@jekyll/build` – Concerned with the Jekyll build pipeline, from reading files, rendering them, and writing them. Includes incremental regeneration.
- `@jekyll/documentation` – Concerned with making jekyllrb.com and other Jekyll documentation great.
- `@jekyll/ecosystem` – Concerned with the fostering of a vibrant ecosystem. Plugins, themes, etc.
- `@jekyll/performance` – Concerned with Jekyll's speed.
- `@jekyll/stability` – Concerned with all things stability, including configuration, defaults, deprecation, etc.
- `@jekyll/windows` – Concerned with Windows compatibility. Can provide testing and guidance for Windows-related bug fixes and enhancements.

Mention one of these teams in a comment below and we'll get this sorted. Thanks!

**Comment by envygeeks** [comment]
This issue is not stale.

**Comment by jekyllbot** [comment]
Hello! We are utilizing a new workflow in our issues and pull requests. Affinity teams have been setup to allow community members to hear about pull requests that may be interesting to them. When a new issue or pull request comes in, we are asking that the author mention the appropriate affinity team. I then assign a random "team captain" or two to the issue who is in charge of triaging it until it is closed or passing it off to another captain. In order to move forward with this new workflow, we need to know: which of the following teams best fits your issue or contribution?

- `@jekyll/build` – Concerned with the Jekyll build pipeline, from reading files, rendering them, and writing them. Includes incremental regeneration.
- `@jekyll/documentation` – Concerned with making jekyllrb.com and other Jekyll documentation great.
- `@jekyll/ecosystem` – Concerned with the fostering of a vibrant ecosystem. Plugins, themes, etc.
- `@jekyll/performance` – Concerned with Jekyll's speed.
- `@jekyll/stability` – Concerned with all things stability, including configuration, defaults, deprecation, etc.
- `@jekyll/windows` – Concerned with Windows compatibility. Can provide testing and guidance for Windows-related bug fixes and enhancements.

Mention one of these teams in a comment below and we'll get this sorted. Thanks!

**Comment by pathawks** [comment]
Looks like we are just waiting on #5052 and #5045?

**Comment by parkr** [comment]
I believe this is now done. :tada: thanks for all the hard work everyone!!!!!
